### PR TITLE
Replace Oracle libjsig.so with OpenJ9 version

### PR DESCRIPTION
**Pull request by andrew-m-leonard**
Signed-off-by: andrew-m-leonard <andrew_m_leonard@uk.ibm.com>

### CODE REVIEW

**Comment by andrew-m-leonard** [comment]
jenkins compile xlinux

File: make/lib/Lib-java.base.gmk [hunk 1]
```diff
@@ -176,6 +182,7 @@ ifeq ($(OPENJDK_TARGET_OS_TYPE), unix)
 
   endif
 endif
+endif
```
Comment (keithc-ca) [inline make/lib/Lib-java.base.gmk]: A comment to help match this with the opening `ifeq` would be helpful.

File: closed/OpenJ9.gmk [hunk 2]
```diff
@@ -179,6 +179,14 @@ $(foreach subdir, j9vm server, \
 		$(subdir)/$(call SHARED_LIBRARY,jvm), \
 		redirector/$(call SHARED_LIBRARY,jvm_jdk11)))
 
+# jsig
+$(call openj9_add_jdk_special, \
+                server/$(call SHARED_LIBRARY,jsig), \
```
Comment (keithc-ca) [inline closed/OpenJ9.gmk]: This should add jsig to the `j9vm` subdirectory (not just the `server` subdirectory.

Please also respect the existing indentation style of this file.

File: closed/OpenJ9.gmk [hunk 3]
```diff
@@ -179,6 +179,14 @@ $(foreach subdir, j9vm server, \
...
+                $(call SHARED_LIBRARY,jsig))
+$(call openj9_add_jdk_special, \
+                $(call SHARED_LIBRARY,jsig), \
```
Comment (keithc-ca) [inline closed/OpenJ9.gmk]: This is redundant and should be removed: jsig is already in the `compressedrefs` (or `default`) subdirectory which is searched before the main `lib` directory.

Comment (keithc-ca) [reply closed/OpenJ9.gmk]: The call on line 183 puts it in the server (and as requested in the j9vm) directories; the same places users will find libjvm.so.

**Comment by andrew-m-leonard** [comment]
Still WIP, having trouble trying to get J9 libjsig.so into right place in image!!

**Comment by andrew-m-leonard** [comment]
I've figured why now, it was because I was working on a more indirect patching of the Lib-java.base.gmk using a "post" extension, however it was causing this issue, hence i'm reverting that.
thanks

File: make/lib/Lib-java.base.gmk (hunk 1 above)
Comment (andrew-m-leonard) [reply make/lib/Lib-java.base.gmk]: added

File: closed/OpenJ9.gmk (hunk 2 above)
Comment (andrew-m-leonard) [reply closed/OpenJ9.gmk]: indentation corrected
subdirs match where libjsig.so was before

**Comment by andrew-m-leonard** [comment]
@keithc-ca @pshipton If you could review again please?

**Comment by andrew-m-leonard** [comment]
jenkins compile xlinux

**Comment by pshipton** [comment]
As @keithc-ca mentioned earlier, to be consistent the libjsig.so also needs to be available in the j9vm directory. This is consistent with having it available in the openjdk server directory (which is only created in openj9 builds for compatibility with openjdk).

Also note that jdk8 will require similar fixes to copy libjsig.so to the correct places.

**Comment by pshipton** [comment]
How do these changes affect Windows, which has a different directory layout?

**Comment by keithc-ca** [comment]
OpenJDK builds for Windows don't have jsig shared libraries (at least not in the latest hotspot builds for jdk8 and jdk11 from https://adoptopenjdk.net/).

**Comment by andrew-m-leonard** [comment]
> 
> 
> As @keithc-ca mentioned earlier, to be consistent the libjsig.so also needs to be available in the j9vm directory. This is consistent with having it available in the server directory (which is only created in openj9 builds for compatibility with openjdk).
> 
> Also note that jdk8 will require similar fixes to copy libjsig.so to the correct places.

I've added it to j9vm.

jdk8 does not have an OpenJDK libjsig.so, so we don't have this issue.

**Comment by andrew-m-leonard** [comment]
Ready again..

**Comment by andrew-m-leonard** [comment]
jenkins compile xlinux

**Comment by andrew-m-leonard** [comment]
jenkins test sanity xlinux

**Comment by pshipton** [comment]
Do these changes affect Windows, which for an OpenJDK build doesn't contain jsig.dll? i.e. does it copy jsig.dll into the server, j9vm, lib directory on Windows?

> jdk8 does not have an OpenJDK libjsig.so, so we don't have this issue.

I believe there is a similar issue, in that the OpenJ9 libjsig.so should be available outside of the OpenJ9 compressedrefs/default directory, since user code must link with this library when using signals. Anyway I can open a new Issue for that.

**Comment by pshipton** [comment]
On Windows the correct directories are bin\server and bin\j9vm and bin. Keith also pointed out the jsig.lib file is also required on Windows, although that could be handled via another PR.

**Comment by keithc-ca** [comment]
Yes, let's defer the Windows issues to a separate PR. I don't think the changes here will negatively affect Windows.

**Comment by pshipton** [comment]
We also need to consider AIX, which uses a libjsig.a file.

**Comment by pshipton** [comment]
Ignore the previous comment, there is no libjsig.a file on AIX.

**Pull request merged**